src/ServantElm.hs

```diff
@@ -166,6 +238,118 @@ mkRequest request =
 
     expect =
       case request ^. reqReturnType of
-        Just _ ->
-          "Http.expectWhatever toMsg"
+        Just elmTypeExpr ->
+          "Http.expectJson" <+> toMsg <+> (typeRefDecoder . definitionToRef) elmTypeExpr
         Nothing -> error "mkHttpRequest: no reqReturnType?"
+
+    body =
+      case request ^. reqBody of
+        Just elmTypeExpr ->
+          "Http.jsonBody" <+> parens ((typeRefEncoder . definitionToRef) elmTypeExpr <+> bodyValue)
+        Nothing ->
+          "Http.emptyBody"
+
+typeRefDecoder :: TypeRef -> Doc ann
+typeRefDecoder (RefCustom TypeName {unTypeName}) = "decode" <> pretty (T.takeWhile (/= ' ') unTypeName)
+typeRefDecoder (RefPrim elmPrim) = case elmPrim of
+  ElmUnit -> parens "JD.map (always ()) (JD.list JD.string)"
+  ElmNever -> parens "JD.fail \"Never is not possible\""
+  ElmBool -> "JD.bool"
+  ElmChar -> "elmStreetDecodeChar"
+  ElmInt -> "JD.int"
+  ElmFloat -> "JD.float"
+  ElmString -> "JD.string"
+  ElmTime -> "Iso.decoder"
+  ElmMaybe t ->
+    parens $
+      "JD.maybe"
+        <+> typeRefDecoder t
+  ElmResult l r ->
+    parens $
+      "elmStreetDecodeEither"
+        <+> typeRefDecoder l
+        <+> typeRefDecoder r
+  ElmPair a b ->
+    parens $
+      "elmStreetDecodePair"
+        <+> typeRefDecoder a
+        <+> typeRefDecoder b
+  ElmTriple a b c ->
+    parens $
+      "elmStreetDecodeTriple"
+        <+> typeRefDecoder a
+        <+> typeRefDecoder b
+        <+> typeRefDecoder c
+  ElmList l -> parens $ "JD.list" <+> typeRefDecoder l
+
+typeRefEncoder :: TypeRef -> Doc ann
```

Comment:
This is taken from `elm-street`, right? You should add a comment, so that we know what we should ask `elm-street` to expose from their api eventually.

Reply:
done